app/handlers.go

```diff
@@ -1,62 +1,118 @@
-		"message": "pong",
-	})
-}
-
-func DownloadHandler(ctx *gin.Context) {
-	fileName := ctx.Param("filename")
-	targetPath, err := file.GetFilePath(ctx.Param("filename"))
-	if err != nil {
-		ctx.String(403, err.Error())
-		return
-	}
-	ctx.Header("Content-Description", "File Transfer")
-	ctx.Header("Content-Transfer-Encoding", "binary")
-	ctx.Header("Content-Disposition", "attachment; filename="+fileName)
-	ctx.Header("Content-Type", "application/octet-stream")
-	ctx.File(targetPath)
-}
-
-func UploadHandler(c *gin.Context) {
-	file, err := c.FormFile("file")
-
-	if err != nil {		
-		log.Fatal(err)
-		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
-            "message": "error receiving file",
-        })
-
-	}
-
-	fileNameSlice := strings.Split(file.Filename, ".")
-	fileExtension := fileNameSlice[len(fileNameSlice) - 1]	
-	newFileName := uuid.New().String()
-
-	err = c.SaveUploadedFile(file, "storage/" + newFileName + "." + fileExtension)
-	if err != nil {
-		log.Fatal(err)
-		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
-            "message": "error saving file",
-        })
-	}
-
-	c.JSON(http.StatusOK, gin.H{
-		"message": fmt.Sprintf("'%s' uploaded", file.Filename),
-		"accessName": newFileName + "." + fileExtension,
-	})
-}
+package app
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"net/http"
+	"net/url"
+	"os"
+	"strconv"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/gujjsPet/image-transform-service/file"
+	"github.com/gujjsPet/image-transform-service/image"
+)
+
+func PingHandler(c *gin.Context) {
+	c.JSON(http.StatusOK, gin.H{
+		"message": "pong",
+	})
+}
+
+func setFileUploadHeaders(c *gin.Context, fileName string) {
+	c.Header("Content-Description", "File Transfer")
+	c.Header("Content-Transfer-Encoding", "binary")
+	c.Header("Content-Disposition", "attachment; filename="+fileName)
+	c.Header("Content-Type", "application/octet-stream")
+}
+
+func DownloadHandler(c *gin.Context) {
+	fileName := c.Param("filename")
+	targetPath, err := file.GetFilePath(c.Param("filename"))
+	if err != nil {
+		c.String(403, err.Error())
+		return
+	}
+
+	setFileUploadHeaders(c, fileName)
+	c.File(targetPath)
+}
+
+func UploadHandler(c *gin.Context) {
+	f, err := c.FormFile("file")
+
+	if err != nil {
+		log.Fatal(err)
+		c.AbortWithStatusJSON(http.StatusBadRequest, gin.H{
+			"message": "error receiving file",
+		})
+
+	}
+
+	newFileName := file.GenerateFilename(f.Filename)
+
+	err = c.SaveUploadedFile(f, "storage/"+newFileName)
+	if err != nil {
+		log.Fatal(err)
+		c.AbortWithStatusJSON(http.StatusInternalServerError, gin.H{
+			"message": "error saving file",
+		})
+	}
+
+	c.JSON(http.StatusOK, gin.H{
+		"message":    fmt.Sprintf("'%s' uploaded", newFileName),
+		"accessName": newFileName,
+	})
+}
+
+func decodeCoordinates(v url.Values) (int, int, int, int) {
+	x0, _ := strconv.Atoi(v.Get("x0"))
+	y0, _ := strconv.Atoi(v.Get("y0"))
+	x1, _ := strconv.Atoi(v.Get("x1"))
+	y1, _ := strconv.Atoi(v.Get("y1"))
+	return x0, y0, x1, y1
+}
```

Comment:
I think it would be better to move this to image.go file

Reply:
This function is parsing the user inputs. My idea for image package (it's single responsibility) was to provide all the image transformations. And I thought this image package should not be responsible for parsing string arguments.

But you are right that those 4 coordinates are strictly connected with images. So I think I will work on integrating it into image package